docs/assignment2.md

```diff
@@ -1 +1,409 @@
+
+*<u>runPlayer</u>*<u>(Player)</u> is what will handle any player. When a *Player* joins the *Game*, *Game* will open a new socket or thread (depending on wether the game is networked or not), and handle that *Player* in the thread/socket. This function calls *Player.getInstruction()*.
+
+*<u>startGame(): Void</u>* initiates the *Game* by declaring user size, *timestamp* and getting the layout. 
+
+The relationships to *Game* are very specific. However, these relations are further described in *Map* and *Player*. At a highlevel, a *Game* can have at most 1 map, but can have any posotive number greater than 1 of *Players*. 
+
+<h5 id="Player">Player</h5>
+
+A *Player* is the class that a User will take control of. The user interacts to the *Player* using the *Parser* [Further described in *Parser*]. A *Player* is a class which also enables the user to explore the world. 
+
+*<u>name: String</u>* is a string which holds the name the user first put in when logging in to VuORK. This must be unique from all other users in the *Game* object. 
+
+*<u>coord: Coordinate</u>* stores where the player currently is in terms of the *Map*. It should be instatiated to the specified *Map.getEntryPoint()*. 
+
+*<u>backpack: Array<BasicItem\></u>* stores references to all of the *Items* the *Player* currently has in his possesion. Instantiated to NULL. It is important to note that this array is of type *BasicItem* and not its superclass, *Item*, because holding *Containers*, would allow for infinite storage. 
+
+*<u>score: Int</u>* is an integer indicator denoting how well the player is playing. It is calculated by taking the number of <u>moves</u> * (number of minutes played) * *health*
+
+*<u>moves: Int</u>* is an integer which counts the number of commands the user has inputted. 
+
+*<u>health: Int</u>* is an integer holding the health of the *Player*. It is between 0 (dead) and 100 (full health).
+
+<u>instruction: String</u> is a String of the last instruction held by the *Player*.
+
+*<u>getUserName(): String</u>* returns <u>name</u>
+
+*<u>getHealth(): Int</u>* returns *health*
+
+*<u>getBackpack(): String</u>* returns a formatted string of items in <u>backpack</u>
+
+<u>movePlayer(coord: Coordinate): Bool</u> will move a player iff the condition *Map.isValidMove(coord: Coordinate)* is true.
+
+*<u>getCurrentPosition(): Coordinate</u>* returns the value of <u>coord</u>
+
+*<u>getInstruction</u>()*, will return the ask *getInstruction* from *Parser* assosciated to the *Player*. 
+
+*<u>setName(String): Void</u>* sets the value of <u>name</u> to String.
+
+The relations of Player are a little complex. A player must contain a *Parser* and a *Printer* (equally, these 2 must can not exist without a *Player*). The *Player* needs these 2 classes to manage I/O individually in a multiplayer scenario. Much like in *Area*, a *Player* object may contain a reference to an *Item*, iff this *Item* does not already exist in an *Area*
```

Comment:
Consider creating a game_engine layer from which both players can interact at the same time. A couple of possible problems, such as syncronization and consistency of the game world, might be mitigated this way